# DP: Use --push-state/--pop-state for gold as well when linking libtsan.

--- a/gcc/gcc.c
+++ b/gcc/gcc.c
@@ -700,10 +700,10 @@
 #define LIBTSAN_SPEC STATIC_LIBTSAN_LIBS
 #elif defined(HAVE_LD_STATIC_DYNAMIC)
 #define LIBTSAN_SPEC "%{static-libtsan:" LD_STATIC_OPTION "}" \
-		     " %{!static-libtsan:%{!fuse-ld=gold:--push-state }--no-as-needed}" \
+		     " %{!static-libtsan:--push-state --no-as-needed}" \
 		     " -ltsan " \
 		     " %{static-libtsan:" LD_DYNAMIC_OPTION "}" \
-		     " %{!static-libtsan:%{fuse-ld=gold:--as-needed;:--pop-state}}" \
+		     " %{!static-libtsan:--pop-state}" \
 		     STATIC_LIBTSAN_LIBS
 #else
 #define LIBTSAN_SPEC "-ltsan" STATIC_LIBTSAN_LIBS
